sktime/forecasting/compose/_multiplexer.py

```diff
@@ -126,45 +126,50 @@ def _check_selected_forecaster(self):
             )
 
     def __or__(self, other):
-        """Magic | method, return MultiplexForecaster.
+        """Magic | (or) method, return (right) concatenated MultiplexForecaster.
+
+        Implemented for `other` being a forecaster, otherwise returns `NotImplemented`.
 
         Parameters
         ----------
-        other : either a forecaster object or a MultiplexForecaster.
-            if forecaster object:
-                add the forecaster to self's list of forecasters.
-            if MultiplexForecaster:
-                create a new MultiplexForecaster with forecasters from both
-                self and other. (Note selected_forecaster of the new
-                MultiplexForecaster will be None, even if it is not None for
-                either self or other)
+        other: `sktime` forecaster, must inherit from BaseForecaster
+            otherwise, `NotImplemented` is returned
 
         Returns
         -------
-        self : returns an instance of self.
+        MultiplexForecaster object, concatenation of `self` (first) with `other` (last).
+            not nested, contains only non-MultiplexForecaster `sktime` forecasters
+        """
+        return self._dunder_concat(
+            other=other,
+            base_class=BaseForecaster,
+            composite_class=MultiplexForecaster,
+            attr_name="forecasters",
+            concat_order="left",
+        )
 
-        Raises
-        ------
-        ValueError if other is not of type MultiplexForecaster or BaseForecaster.
+    def __ror__(self, other):
```

Comment:
So, I could be wrong, but do we actually need this  `__ror__` case?  Why I ask - in theory `__ror__` covers 2 legitimate cases:

- 1 `BaseForecaster` `|` `MultiplexForecaster`
- 2 `MultiplexForecaster` `|` `MultiplexForecaster`

But I think both these cases are already covered!  

1 - should be handled by the `__or__` method defined in `BaseForecaster`. In this case we would convert the base forecaster to a multiplex forecaster, then pass it to the multiplex forecaster `|` dunder. (so it essentially becomes case 2)
2 - can be handled by the fact that the object on the left is already is a `MultiplexForecaster` and thus is covered by `__or__` already.

I guess in theory the `__ror__` could be helpful for raising `NotImplemented` if we tried to do something like this:

"this shouldn't work" | `MultiplexForecaster`

But I would imagine this would already be raised without us needed to explicitly define it.  I also worries it will be confusing/raise bugs to have this unnecessary `__ror__`/I don't completely understand which dunder would take precedent in the case that multiple could handle the case at hand...  (ie, could be fun to see in the case of `MultiplexForecaster` `|` `MultiplexForecaster` if `__or__` or `__ror__` gets called. (my guess is `__or__` but might play around later with the debugger to see!)) 

Please let me know if you feel differently though/that `__ror__` is needed.

Reply:
As a follow up - I guess I would have similar questions about `__radd__` and `__rmul__`, though I am less familiar with whether these bases are fully covered already (though one might imagine that if there were not a similar type casting could happen at the `BaseForecaster` or `BaseTransformer` stage.

Reply:
> So, I could be wrong, but do we actually need this  `__ror__` case?

Two reasons:

* In the dunders I wrote, I also allowed the named tuple case, e.g.,
`("tf1", trafo) * ("tf2", trafo2) * forecaster`
which works as long as the "special" element (e.g., forecaster, classifier) is not a tuple.
One could argue, one does not want this, though

* this is the "real" reason imo that makes the right function necessary, imagine the forecaster pipeline `trafo1 * trafo2 * forecaster * postproc` - how would this work if you do not want to attach to the transformers forecaster specific functionality? It can only work via the `__rmul__` of `forecaster`.
The alternative would require one to add more and more case distinctions to `transformer.__mul__` which violates the open/closed principle.

Reply:
so for the "almost commutative" compositions which operate only on classes of the same type, I guess one could get away with the left dunder only. But, if you have type inhomogeneous ones, I think you need the right dunder.

Reply:
Hmm, I had initially also wanted to support tuples, though in the end I couldn't see a good way to have something like `("ets", AutoETS()) | ("naive", NaiveForecaster)` end up as a `MultiplexForecaster`, and to me the requirement that you need at least one forecaster/`MultiplexForecaster` seemed like it might be a weird requirement to the user. (and might cause more confusion than it saved, so I decided to limit myself to only supporting `BaseForecaster` and `MultiplexForecaster` as inputs).  So personally I might be in the camp of not wanting this, at least for the case of the `MultiplexForecaster`...

As to the second point, I am still not sure that this applies to `MultiplexForecaster` though/the need for an `__ror__` though? Also - it could just be me (and probably is just me) but as soon as there are multiple things one could want from  `trafo1 * trafo2 * forecaster * postproc` I worry that having the dunders handle this automatically gets a bit opaque/confusing?  For example, I read it as - make a `TransformedTargetForecaster` and then apply some post processing, but as soon as there are multiple ways to do this and the dunder is handling it automatically (ie either attaching or not attaching the transformers forecaster specific functionality) it gets a bit confusing/I wonder if we should support this.  (again - probably ignorance on my part that is leading to this confusing, so in that case apologies!)

Reply:
> but as soon as there are multiple things one could want from  `trafo1 * trafo2 * forecaster * postproc` 

Quick question - what are the multiple things one could want, i.e., why do you think this is ambiguous?

Reply:
Sorry - looking back I realized I just misread this comment:
> how would this work if you do not want to attach to the transformers forecaster specific functionality?
And thought you were saying something else.  So now I am not thinking that you want multiple things. 

However... Upon re-reading I think now I potentially have a solution for you - if you have the `__mul__` in `BaseTransformer` simply create a new `TransformedTargetForecaster`, and then pass it off to the `__mul__` in `TransformedTargetForecaster` to actually create the new object (ie akin to what I do in `BaseForecaster` for the `|` dunder), then you could have this work without having to attach to the transformer forecaster specific functionality? Thoughts? :)

Reply:
No, that would not solve it, because then the `TransformedTargetForecaster` creation logic would be attached to `BaseTransformer`? So, forecaster logic being attached to all transformers.

Reply:
Yes, I see your point/if you want to keep references to forecasters out of `BaseTransfromer` this solution makes sense. 

Though I still don't think we need `__ror__`. :)

Reply:
I think we need it in `BaseForecaster`, but we might not need it in `BaseTransformer`, you might be right. Although I like to play it safe in cases like this, e.g., end a case distinction with `else raise ValueError("unreachable state")`. You could call that (over?) defensive programming https://en.wikipedia.org/wiki/Defensive_programming